content/r-package-dev/Documentation.Rmd

```diff
@@ -0,0 +1,253 @@
+
+Once you build your `.Rd` files, you will see them appear in the `man` folder. Each function will have it's own `.Rd`. You should now be able to use `?functionName` and see your help file. For the example we have been following, we should be able to see our help file by executing `?is.valid.pH`.
+
+## Vignettes
+
+Vignettes are PDF or HTML files made available to your users when they execute `help` and specify a package name, e.g. `help(package="dataRetrieval")`. Vignettes are typically used to explain information about the package that does not fit into the individual function documentation, such as common usages and workflows, background information, or statistical methods and references. It's a good rule of thumb to include at least one vignette to illustrate a typical workflow that someone might expect to use with your package.
+
+Vignettes are created using [R Markdown](http://rmarkdown.rstudio.com/) files and the `rmarkdown` package. This course will not teach how to use R Markdown, so please refer to the following links for reference:
+
+- [R Markdown quick tour](http://rmarkdown.rstudio.com/authoring_quick_tour.html)
+- [R Markdown overview](http://rmarkdown.rstudio.com/lesson-1.html)
+- [R Markdown cheatsheet](https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf)
+- [R Markdown examples](http://rmarkdown.rstudio.com/gallery.html)
+
+To start, create a top-level directory called `vignettes` in your package. Using the `devtools` function `use_vignettes`, you can set up the vignette template. Try it by executing `devtools::use_vignettes("my-vignette")`. You should now see an `.Rmd` (R Markdown) file in your new `vignettes` folder with the name "my-vignette". When you open the file, you will see that the header includes the following:
+
+```{r eval=FALSE}
+title: "Vignette Title"
+author: "Vignette Author"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+```
+
+Including `Sys.Date()` in the date field is optional but highly recommended because anytime you update the vignette, it will automatically update. The default vignette also includes a lot of information about vignettes, so don't forget to delete this to make room for your content.
+
+In the body of the R Markdown document, add a code chunk that loads the `rmarkdown` package (`library(rmarkdown)`). Populate the rest of the document as you would any other R Markdown file. Remember, if you are unfamiliar with R Markdown, visit the list of references from above. 
+
+Save your new R Markdown file. The next time you build your package, your vignette should now be available!
+
+**NEED `VignetteBuilder: knitr` or `BuildVignettes: true` in DESCRIPTION ?!?! **
```

Comment:
@ldecicco-USGS I saw these fields in dataRetrieval. Are they required? What do they do?